### PR TITLE
EUI-3594: Error message specifics not being displayed

### PR DESCRIPTION
**Pull request by Daniel-Lam**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-3594

### Change description ###
Fix issue with HTTP error handling post-Angular `HttpClient` upgrade, which failed to map error properties between new `HttpErrorResponse` objects being returned and `HttpError` domain model objects for UI display.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/shared/components/case-viewer/case-viewer.component.spec.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import { HttpErrorResponse } from '@angular/common/http';
```
Comment (paul-graham) [inline src/shared/components/case-viewer/case-viewer.component.spec.ts:1]: I'm seeing an `xdescribe(...)` for all the tests in this suite...?

**Review by paul-graham — CHANGES_REQUESTED** [review]
Just that one question in `case-viewer.component.spec.ts` where it looks like there are changes but all the tests are actually being skipped. Not sure if you were aware of it, @Daniel-Lam, so this is to flag it up for your attention rather than _actually_ request changes.

File: src/shared/components/case-viewer/case-viewer.component.spec.ts (hunk 1 above)
Comment (Daniel-Lam) [reply src/shared/components/case-viewer/case-viewer.component.spec.ts:1]: @paul-graham - that's a bit odd... I tried an `fdescribe` and all those tests fail! They ought to be fixed but that would be outside the scope of this PR, so I've left as `xdescribe` (as it was) for now.

Comment (paul-graham) [reply src/shared/components/case-viewer/case-viewer.component.spec.ts:1]: Okay, cool. So long as you're aware of it. We should create a tech debt ticket to tidy up any tests that have an `x` prefix (`xdescribe` or `xit`) - either fix them or remove them. 😄

**Review by paul-graham — APPROVED** [review]
The `xdescribe` hasn't been introduced for this PR and @Daniel-Lam was aware it was there so I'm happy to approve these changes.

**Review by adnanakgun — APPROVED** [review]

**Pull request merged**